src/Internal/Bitwise.php

```diff
@@ -24,78 +24,33 @@
      */
     public static function bitwise(BigInteger $x, BigInteger $y, string $operator) : BigInteger
     {
-        $x2 = $x->abs()->toBase(2);
-        $y2 = $y->abs()->toBase(2);
+        $binaryX = self::toBinary($x);
+        $binaryY = self::toBinary($y);
 
-        // prefix a 0 to ensure than when applying the two's complement, length in bits will remain the same
-        $x2 = '0' . $x2;
-        $y2 = '0' . $y2;
+        $maxLength = max(strlen($binaryX), strlen($binaryY));
 
-        $lx = strlen($x2);
-        $ly = strlen($y2);
-
-        $length = max($lx, $ly);
-
-        if ($lx < $length) {
-            $x2 = str_repeat('0', $length - strlen($x2)) . $x2;
-        }
-
-        if ($ly < $length) {
-            $y2 = str_repeat('0', $length - strlen($y2)) . $y2;
-        }
-
-        if ($x->isNegative()) {
-            $x2 = self::twosComplement($x2);
-        }
-
-        if ($y->isNegative()) {
-            $y2 = self::twosComplement($y2);
-        }
+        $binaryX = $binaryX[0] . str_pad(substr($binaryX, 1), $maxLength - 1, $binaryX[0], STR_PAD_LEFT);
```

Comment:
These two lines are the only change to my original implementation, the rest is just removing your bit-string prototype.